pennylane/templates/embeddings/qaoaembedding.py

```diff
@@ -172,30 +172,36 @@ def __init__(self, features, weights, wires, local_field="Y", do_queue=True, id=
 
         shape = qml.math.shape(features)
 
-        if len(shape) != 1:
-            raise ValueError(f"Features must be a one-dimensional tensor; got shape {shape}.")
+        if len(shape) not in {1, 2}:
+            raise ValueError(
+                "Features must be a one-dimensional tensor or two-dimensional with broadcasting;"
+                f" got shape {shape}."
+            )
 
-        n_features = shape[0]
+        n_features = shape[-1]
```

Comment:
Does this also imply a breaking change? (Likely further occurrences in the PR)

Reply:
No, see above :)